tests/test_tools.py

```diff
@@ -14,11 +14,128 @@ def test_angular_avg_nd_3():
     P = np.repeat(P, 100).reshape(400, 400, 100)
     freq = [x, x, np.linspace(-2, 2, 100)]
     p_k, k_av_bins = angular_average_nd(P, freq, bins=50, n=2)
+    p_k_lin, k_av_bins_lin = angular_average_nd(
+        P, freq, bins=50, n=2, interpolation_method="linear"
+    )
     print(p_k[6:, 0], k_av_bins[6:] ** -2.0)
     assert (
         np.max(np.abs((p_k[6:, 0] - k_av_bins[6:] ** -2.0) / k_av_bins[6:] ** -2.0))
         < 0.05
     )
+    assert (
+        np.max(
+            np.abs(
+                (p_k_lin[6:, 0] - k_av_bins_lin[6:] ** -2.0) / k_av_bins_lin[6:] ** -2.0
+            )
+        )
+        < 0.05
+    )
+
+
+def test_interp_w_weights():
+    x = np.linspace(-3, 3, 40)
+    P = np.ones((40, 40, 40))
+    P[:4, 3:6, 7:10] = 0
+    weights = np.ones_like(P)
+    weights[:4, :, :] = 0
+    weights[:, 3:6, :] = 0
+    weights[:, :, 7:10] = 0
+
+    # Test 4D avg works
+    freq = [x, x, x]
+    p_k_lin, k_av_bins_lin = angular_average(
+        P,
+        freq,
+        bins=10,
+        interpolation_method="linear",
+        weights=weights,
+        angular_resolution=0.1,
+    )
+
+    assert np.all(p_k_lin == 1.0)
```

Comment:
this is a nice test... you could also parameterize it by sending in arrays from 1D up to say 4D, and a couple different angular resolutions.